src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -537,7 +537,7 @@ public void startActivityForResult(Intent intent, int requestCode) {
     private String checkInputParameters(String resource, String clientId, String redirectUri,
             PromptBehavior behavior, AuthenticationCallback<AuthenticationResult> callback) {
         if (mContext == null) {
-            throw new AuthenticationException(ADALError.DEVELOPER_CONTEXT_IS_NOT_PROVIDED);
+            throw new IllegalArgumentException("context", new AuthenticationException(ADALError.DEVELOPER_CONTEXT_IS_NOT_PROVIDED));
```

Comment:
I don't actually need inner exception, just left it here for long memory)
And it's actually not breaking change, because this method is being called from async methods, which throws it is as RuntimeException (not returned in callback)